Core/main.cpp

```diff
@@ -16,14 +17,34 @@
 
 #include "Renderer.h"
 #include "Primitives.h"
+#include "System.h"
+
+
+
+
+class Core {
+private:
+	std::vector<System*> systems;
+	
+public:
+	void AddSystem(System* system) {
```

Comment:
For future note: we should have a separate function to activate a system, we may want to register something that doesn't immediately run